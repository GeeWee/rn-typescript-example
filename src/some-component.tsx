import React from 'react';
import {
	AppRegistry,
	Text,
	View
} from 'react-native';

export const SomeComponent = () => {
	return (
	<View
		style={{
			flex: 1, justifyContent: 'center', alignItems: 'center'
		}}
	>
		<Text>Hello, Typed world</Text>
	</View>
	)
};