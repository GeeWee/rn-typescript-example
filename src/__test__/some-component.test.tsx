import 'jest';
import 'react-native';
import React from 'react';
import { SomeComponent } from '../some-component';
import renderer from 'react-test-renderer';

describe('Basic breakpoints and sourcemaps', () => {
	it('Should show you the value of the variables the whole way through', () => {
		const x = 2;
		let y = 3;
		debugger;
		let z = x + y;
		expect(z).toEqual(5);
	});
});

describe('Some component', () => {
	it('Should import', () => {
		const component = SomeComponent;
		expect(component).toBeDefined();
	});
	
	
	it('Should render', () => {
		const renderedComponent = renderer.create(<SomeComponent />).toJSON();
		expect(renderedComponent).toMatchSnapshot();
	});
});