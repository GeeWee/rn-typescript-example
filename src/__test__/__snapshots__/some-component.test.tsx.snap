// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Some component Should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Hello, Typed world
  </Text>
</View>
`;
